cli/cmd/send.go

```diff
@@ -3,91 +3,101 @@ package cmd
 import (
 	"bugless/shared"
 	"context"
+	"encoding/json"
 	"log"
 	"math/big"
+	"os/exec"
 	"time"
 
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/gligneul/eggroll"
-	"github.com/gligneul/eggroll/eggeth"
-	"github.com/gligneul/eggroll/eggtypes"
+	"github.com/gligneul/rollmelette"
 	"github.com/spf13/cobra"
 )
 
 var sendArgs struct {
-	accountIndex uint32
+	fromAddress string
 }
 
 var sendCmd = &cobra.Command{
 	Use:   "send",
 	Short: "Send an advance-state input to the contract",
 }
 
-func sendDo(send func(context.Context, *eggroll.Client, eggeth.Signer) (int, error)) {
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-	defer cancel()
+var addressBook = rollmelette.NewAddressBook()
 
-	client, signer, err := eggroll.NewDevClient(ctx, shared.Codecs())
+var dappAddress = "0x70ac08179605AF2D9e75782b8DEcDD3c22aA4D0C"
+
+func sendDo(inputKind shared.InputKind, payload any, send func(string, context.Context) ([]byte, error)) {
+	payloadJson, err := json.Marshal(payload)
 	if err != nil {
 		log.Fatal(err)
 	}
-
-	if sendArgs.accountIndex != 0 {
-		err := signer.(*eggeth.MnemonicSigner).SetAccount(sendArgs.accountIndex)
-		if err != nil {
-			log.Fatal(err)
-		}
+	input := shared.Input{
+		Kind:    inputKind,
+		Payload: payloadJson,
 	}
-
-	inputIndex, err := send(ctx, client, signer)
-
+	inputJson, err := json.Marshal(input)
 	if err != nil {
 		log.Fatal(err)
 	}
-	log.Printf("added input %v", inputIndex)
+	inputJsonStr := string(inputJson)
 
-	result, err := client.WaitFor(ctx, inputIndex)
```

Comment:
Hey @gligneul 
Do you think something like [waitForInput](https://github.com/gligneul/rollmelette/blob/f3586a670dbaff2cbc015ed60afa7335d20dc72f/integration/nonodo_test.go#L100) function is a good helper function to replace the `WaitFor` function above?

Similarly, [to implement the stateRun function](https://github.com/crypto-bug-hunters/bug-less/blob/afa5b711ce4269bc67a5255430c03903fea90387/cli/cmd/state.go#L31), seems this [getNodeState](https://github.com/gligneul/rollmelette/blob/f3586a670dbaff2cbc015ed60afa7335d20dc72f/integration/genqlient.go#L392C6-L392C18) function  would be a good helper, but it's not exported.